core/src/main/java/com/sequenceiq/cloudbreak/service/StackCommonService.java

```diff
@@ -160,10 +161,29 @@ public void putStopInWorkspace(String name, Long workspaceId) {
         put(stack, updateStackJson);
     }
 
-    public void putSyncInWorkspace(String name, Long workspaceId) {
+    public void putStopInWorkspaceByCrn(String crn, Long workspaceId) {
         User user = userService.getOrCreate(restRequestThreadLocalService.getCloudbreakUser());
         permissionCheckingUtils.checkPermissionByWorkspaceIdForUser(workspaceId, WorkspaceResource.STACK, ResourceAction.WRITE, user);
-        Stack stack = stackService.getByNameInWorkspace(name, workspaceId);
+        Stack stack = stackService.getByNameInWorkspace(crn, workspaceId);
+        MDCBuilder.buildMdcContext(stack);
+        if (!cloudParameterCache.isStartStopSupported(stack.cloudPlatform())) {
+            throw new BadRequestException(String.format("Stop is not supported on %s cloudplatform", stack.cloudPlatform()));
+        }
+        UpdateStackV4Request updateStackJson = new UpdateStackV4Request();
+        updateStackJson.setStatus(StatusRequest.STOPPED);
+        updateStackJson.setWithClusterEvent(true);
+        put(stack, updateStackJson);
+    }
+
+    public void putSyncInWorkspace(String name, String crn, Long workspaceId) {
+        User user = userService.getOrCreate(restRequestThreadLocalService.getCloudbreakUser());
+        permissionCheckingUtils.checkPermissionByWorkspaceIdForUser(workspaceId, WorkspaceResource.STACK, ResourceAction.WRITE, user);
+        Stack stack;
+        if (!Strings.isNullOrEmpty(name)) {
```

Comment:
I know that this is probably a quick solution but I think the request should fail if 2 mutually exclusive parameters are supplied.